### PR TITLE
Feature/nodea hostname label

### LINKED ISSUE #1033: Chore: Lack of node labels for distinction.
### Description

Nodes in a cluster build by Claudie do not contains any label with unique value. Most of the time the label with unique value is `kubernetes.io/hostname`, which Claudie currently doesn't support.
This might be an issue, when trying to apply Node Affinity, Pod Topology Spread Constraints or anything else, which requires clear uniqueness of the nodes.

In the following code space you can see all labels for each node in cluster, which was created by the InputManifest in **How to reproduce** section.

```
Labels:             beta.kubernetes.io/arch=amd64
                    beta.kubernetes.io/os=linux
                    claudie.io/node-type=compute
                    claudie.io/nodepool=jkb-gcp-compute-ehve9n9
                    claudie.io/provider=gcp
                    claudie.io/provider-instance=gcp-1
                    kubernetes.io/arch=amd64
                    kubernetes.io/os=linux
                    topology.kubernetes.io/region=europe-west2
                    topology.kubernetes.io/zone=europe-west2-a
                    v1.kubeone.io/operating-system=ubuntu
Labels:             beta.kubernetes.io/arch=amd64
                    beta.kubernetes.io/os=linux
                    claudie.io/node-type=compute
                    claudie.io/nodepool=jkb-gcp-compute-ehve9n9
                    claudie.io/provider=gcp
                    claudie.io/provider-instance=gcp-1
                    kubernetes.io/arch=amd64
                    kubernetes.io/os=linux
                    topology.kubernetes.io/region=europe-west2
                    topology.kubernetes.io/zone=europe-west2-a
                    v1.kubeone.io/operating-system=ubuntu
Labels:             beta.kubernetes.io/arch=amd64
                    beta.kubernetes.io/os=linux
                    claudie.io/node-type=control
                    claudie.io/nodepool=jkb-gcp-control-exvmvv7
                    claudie.io/provider=gcp
                    claudie.io/provider-instance=gcp-1
                    kubernetes.io/arch=amd64
                    kubernetes.io/os=linux
                    topology.kubernetes.io/region=europe-west2
                    topology.kubernetes.io/zone=europe-west2-a
                    v1.kubeone.io/operating-system=ubuntu
```

### How to reproduce

Run `kubectl apply` with the following InputManifest in our `e2e` cluster.

``` yaml
apiVersion: claudie.io/v1beta1
kind: InputManifest
metadata:
  name: jakub-test
spec:
  providers:
    - name: gcp-1
      providerType: gcp
      secretRef:
        name: gcp-secret
        namespace: e2e-secrets
  nodePools:
    dynamic:
    - name: jkb-gcp-control
      providerSpec:
        name: gcp-1
        region: europe-west2
        zone: europe-west2-a
      count: 1
      serverType: e2-small
      image: ubuntu-os-cloud/ubuntu-2204-jammy-v20221206
      storageDiskSize: 50
    - name: jkb-gcp-compute
      providerSpec:
        name: gcp-1
        region: europe-west2
        zone: europe-west2-a
      count: 2
      serverType: e2-small
      image: ubuntu-os-cloud/ubuntu-2204-jammy-v20221206
      storageDiskSize: 50
  kubernetes:
    clusters:
      - name: jakub-test
        version: v1.26.0
        network: 192.168.2.0/24
        pools:
          control:
            - jkb-gcp-control
          compute:
            - jkb-gcp-compute
```

### Exit criteria

- [ ] Add kubernetes.io/hostname or any other label, which can be used as a unique identifier for nodes

### PR DESCRIPTION
**Pull request by cloudziu**
Closes #1033

### CODE REVIEW

File: services/kuber/server/domain/utils/nodes/patch.go [hunk 1]
```diff
@@ -72,15 +72,17 @@ func (p *Patcher) PatchLabels() error {
...
-		}
 		for _, node := range np.Nodes {
 			nodeName := strings.TrimPrefix(node.Name, fmt.Sprintf("%s-", p.clusterID))
-			if err1 := p.kc.KubectlPatch("node", nodeName, patchPath, "--type", "json"); err1 != nil {
-				p.logger.Err(err1).Str("node", nodeName).Msgf("Failed to patch labels on node with path %s", patchPath)
-				err = fmt.Errorf("error while patching one or more nodes with labels")
+			for key, value := range l {
+				patchPath, err1 := buildJSONPatchString("replace", "/metadata/labels/" + key, value)
+				if err1 != nil {
+					return fmt.Errorf("failed to create label %s patch path for %s : %w", key, np.Name, err)
```
Comment (Despire) [inline services/kuber/server/domain/utils/nodes/patch.go:80]: Also include the `err1`, you can use multiple `%w` for wrapping multiple errors

File: services/kuber/server/domain/utils/nodes/patch.go [hunk 2]
```diff
@@ -72,15 +72,17 @@ func (p *Patcher) PatchLabels() error {
 			return fmt.Errorf("failed to create labels for %s : %w", np.Name, err)
```
Comment (Despire) [inline services/kuber/server/domain/utils/nodes/patch.go:72]: Also include the `err1` when returning.

**Review by Despire — CHANGES_REQUESTED** [review]
Looks Great! 👍  One (maybe mistake) that I've noticed

**Review by Despire — APPROVED** [review]

File: services/kuber/server/domain/utils/nodes/patch.go [hunk 3]
```diff
@@ -69,18 +69,20 @@ func (p *Patcher) PatchLabels() error {
 	for _, np := range p.desiredNodepools {
 		l, err1 := nodes.GetAllLabels(np, nil)
```
Comment (JKBGIT1) [inline services/kuber/server/domain/utils/nodes/patch.go:70]:
```suggestion
		allLabels, err1 := nodes.GetAllLabels(np, nil)
```
Just a suggestion to make the variable name more self-explanatory.

File: services/kuber/server/domain/utils/nodes/patch.go [hunk 4]
```diff
@@ -69,18 +69,20 @@ func (p *Patcher) PatchLabels() error {
...
-		patchPath, err1 := buildJSONPatchString("replace", "/metadata/labels", l)
-		if err1 != nil {
-			return fmt.Errorf("failed to create labels patch path for %s : %w", np.Name, err)
-		}
 		for _, node := range np.Nodes {
 			nodeName := strings.TrimPrefix(node.Name, fmt.Sprintf("%s-", p.clusterID))
-			if err1 := p.kc.KubectlPatch("node", nodeName, patchPath, "--type", "json"); err1 != nil {
-				p.logger.Err(err1).Str("node", nodeName).Msgf("Failed to patch labels on node with path %s", patchPath)
-				err = fmt.Errorf("error while patching one or more nodes with labels")
+			for key, value := range l {
```
Comment (JKBGIT1) [inline services/kuber/server/domain/utils/nodes/patch.go:77]:
```suggestion
			for key, value := range allLabels {
```
This just follows the previous suggestion.

**Review by JKBGIT1 — APPROVED** [review]
LGTM :+1:  I just left a suggestion, which you can implement in case you like it.

**Pull request merged**